### ISSUE TITLE
Segmentation fault in Dart 2.17.3 when using enhanced enum under Linux

### ISSUE THREAD

**Issue by JKRhb** [dart-lang/sdk#49296]
Using a library where I recently refactored code to use enhanced enums I am getting a segmentation fault in my CI pipeline (both on GitLab and GitHub) using Dart 2.17.3 under Linux. The enum that seems to cause the error can be found [here](https://github.com/shamblett/coap/blob/8899dd6829bd5f8604d96bc7701095f44b38c941/lib/src/coap_option_type.dart#L59). See the log of the crash below:

```
===== CRASH =====
si_signo=Segmentation fault(11), si_code=1, si_addr=0x18
version=2.17.3 (stable) (Wed Jun 1 11:06:41 2022 +0200) on "linux_x64"
pid=1758, thread=1794, isolate_group=main(0x558d6d192000), isolate=test/binding_coap/binding_coap_test.dart(0x558d6d192800)
isolate_instructions=558d6b78c420, vm_instructions=558d6b78c420
  pc 0x0000558d6be39c5e fp 0x00007f3c0967c1f0 dart::kernel::ScopeBuilder::LookupVariable(long)+0x1fe
  pc 0x0000558d6be39a1f fp 0x00007f3c0967c210 dart::kernel::ScopeBuilder::VisitVariableGet(long)+0xf
  pc 0x0000558d6be37970 fp 0x00007f3c0967c250 dart::kernel::ScopeBuilder::VisitExpression()+0x610
  pc 0x0000558d6be382c8 fp 0x00007f3c0967c320 dart::kernel::ScopeBuilder::VisitConstructor()+0x228
  pc 0x0000558d6be363ad fp 0x00007f3c0967c4a0 dart::kernel::ScopeBuilder::BuildScopes()+0x97d
  pc 0x0000558d6b9ee0a7 fp 0x00007f3c0967c7e0 dart::ParsedFunction::EnsureKernelScopes()+0x37
  pc 0x0000558d6bdffc13 fp 0x00007f3c0967c810 dart::kernel::StreamingFlowGraphBuilder::ParseKernelASTFunction()+0x53
  pc 0x0000558d6bdffa71 fp 0x00007f3c0967c8c0 dart::kernel::StreamingFlowGraphBuilder::BuildGraph()+0xb1
  pc 0x0000558d6be136e5 fp 0x00007f3c0967cb80 dart::kernel::FlowGraphBuilder::BuildGraph()+0x65
  pc 0x0000558d6ba890a0 fp 0x00007f3c0967cdc0 dart::DartCompilationPipeline::BuildFlowGraph(dart::Zone*, dart::ParsedFunction*, dart::ZoneGrowableArray<dart::ICData const*>*, long, bool)+0x40
  pc 0x0000558d6ba8a0c2 fp 0x00007f3c0967d4d0 dart::CompileParsedFunctionHelper::Compile(dart::CompilationPipeline*)+0x4d2
  pc 0x0000558d6ba8ad3c fp 0x00007f3c0967d6e0 dart+0x21e3d3c
  pc 0x0000558d6ba8b[12](https://github.com/namib-project/dart_wot/runs/6964529716?check_suite_focus=true#step:7:15)5 fp 0x00007f3c0967d720 dart::Compiler::EnsureUnoptimizedCode(dart::Thread*, dart::Function const&)+0xb5
  pc 0x0000558d6ba64a14 fp 0x00007f3c0967d770 dart::SourceReport::VisitFunction(dart::JSONArray*, dart::Function const&)+0x[13](https://github.com/namib-project/dart_wot/runs/6964529716?check_suite_focus=true#step:7:16)4
  pc 0x0000558d6ba64eae fp 0x00007f3c0967d800 dart::SourceReport::VisitLibrary(dart::JSONArray*, dart::Library const&)+0x12e
  pc 0x0000558d6ba6533f fp 0x00007f3c0967d8a0 dart::SourceReport::PrintJSON(dart::JSONStream*, dart::Script const&, dart::TokenPosition, dart::TokenPosition)+0x12f
  pc 0x0000558d6ba5551f fp 0x00007f3c0967d9f0 dart+0x21ae51f
  pc 0x0000558d6ba4c261 fp 0x00007f3c0967db40 dart::Service::InvokeMethod(dart::Isolate*, dart::Array const&, bool)+0x321
  pc 0x0000558d6ba4c9b1 fp 0x00007f3c0967db60 dart::Service::HandleIsolateMessage(dart::Isolate*, dart::Array const&)+0x11
  pc 0x0000558d6b92d5a8 fp 0x00007f3c0967dc40 dart::IsolateMessageHandler::HandleMessage(std::__2::unique_ptr<dart::Message, std::__2::default_delete<dart::Message> >)+0x278
  pc 0x0000558d6b95650d fp 0x00007f3c0967dcb0 dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)+0x[14](https://github.com/namib-project/dart_wot/runs/6964529716?check_suite_focus=true#step:7:17)d
  pc 0x0000558d6b956bef fp 0x00007f3c0967dd00 dart::MessageHandler::TaskCallback()+0x1df
  pc 0x0000558d6ba78078 fp 0x00007f3c0967dd80 dart::ThreadPool::WorkerLoop(dart::ThreadPool::Worker*)+0x148
  pc 0x0000558d6ba784cd fp 0x00007f3c0967ddb0 dart::ThreadPool::Worker::Main(unsigned long)+0x6d
  pc 0x0000558d6b9ed328 fp 0x00007f3c0967de70 dart+0x[21](https://github.com/namib-project/dart_wot/runs/6964529716?check_suite_focus=true#step:7:24)463[28](https://github.com/namib-project/dart_wot/runs/6964529716?check_suite_focus=true#step:7:31)
-- End of DumpStackTrace
=== Crash occured when compiling package:coap/src/coap_option_type.dart_OptionType_OptionType. in unoptimized JIT mode in unknown pass
=== Flow Graph not available
```

**Comment by alexmarkov** [comment]
This looks similar to https://github.com/flutter/flutter/issues/103656 and probably fixed in 539cccd8234ee10fddca244d9855ce0e0e5d8c45.
/cc @liamappelbe

**Comment by JKRhb** [comment]
Thank @alexmarkov and @liamappelbe, that must also be the cause for this issue, as the enum has in fact a `name` field :) As the problem has already been fixed, I will close the issue then :)

**Comment by JKRhb** [comment]
Hi @liamappelbe, just a quick question: Do you think the fix will become part of a Dart 2.17.6? Or will it only be included in 2.18?

**Comment by KOGI** [comment]
Can confirm this issue still exists in `2.17.6`
